qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -81,75 +92,75 @@ def __init__(
         self.ansatz = ansatz
         self.initial_state = initial_state
-        self._vqe = VQE(
-            ansatz=None,
-            quantum_instance=self._quantum_instance,
-            optimizer=self._optimizer,
-            initial_point=self._initial_point,
-            gradient=self._gradient,
-            expectation=self._expectation,
-            include_custom=self._include_custom,
-        )
+        self.callback = callback
 
     @property
     def quantum_instance(self) -> QuantumInstance:
         """Getter of the quantum instance."""
-        return self._quantum_instance
+        return self._vqe.quantum_instance
 
     @quantum_instance.setter
     def quantum_instance(self, q_instance: QuantumInstance) -> None:
         """Setter of the quantum instance."""
-        self._quantum_instance = q_instance
+        self._vqe.quantum_instance = q_instance
 
     @property
     def optimizer(self) -> Optional[Optimizer]:
         """Getter of the optimizer."""
-        return self._optimizer
+        return self._vqe.optimizer
 
     @optimizer.setter
     def optimizer(self, optimizer: Optional[Optimizer]) -> None:
         """Setter of the optimizer."""
-        self._optimizer = optimizer
+        if optimizer is None:
+            return
+        self._vqe.optimizer = optimizer
 
     @property
     def initial_point(self) -> Optional[np.ndarray]:
         """Getter of the initial point."""
-        return self._initial_point
+        return self._vqe.initial_point
 
     @initial_point.setter
     def initial_point(self, initial_point: Optional[np.ndarray]) -> None:
         """Setter of the initial point."""
-        self._initial_point = initial_point
+        if initial_point is None:
+            return
+        self._vqe.initial_point = initial_point
 
     @property
     def gradient(self) -> Optional[Union[GradientBase, Callable]]:
         """Getter of the gradient function"""
-        return self._gradient
+        return self._vqe.gradient
 
     @gradient.setter
     def gradient(self, gradient: Optional[Union[GradientBase, Callable]]) -> None:
         """Setter of the gradient function"""
-        self._gradient = gradient
+        if gradient is None:
+            return
+        self._vqe.gradient = gradient
 
     @property
     def expectation(self) -> Optional[ExpectationBase]:
         """Getter of the expectation."""
-        return self._expectation
+        return self._vqe.expectation
 
     @expectation.setter
     def expectation(self, expectation: Optional[ExpectationBase]) -> None:
         """Setter of the expectation."""
-        self._expectation = expectation
+        if expectation is None:
+            return
+        self._vqe.expectation = expectation
 
     @property
     def include_custom(self) -> bool:
         """Getter of the ``include_custom`` setting for the ``expectation`` setting."""
-        return self._include_custom
+        return self._vqe._include_custom
```

Comment:
This is now accessing a private member `_include_custom` of VQE. We should figure something out here - maybe either delay instantiation of VQE until its needed in solve - or perhaps add public setter/getters to VQE. We should avoid using members of VQE that are marked as private.

Reply:
Yes, I am aware of this problem. I have been meaning to open an issue on Terra to address the addition of missing property setters and to resolve the problem I mentioned above. I have simply not gotten around to opening the issue.
Because this fix is needed for the upcoming Quantum challenge I would like to merge this PR as is and address this once these shortcomings are resolved in Terra.

I can add a TODO comment in all relevant places if you like.

Reply:
By creating VQE in solve (where these setters here just stored the vars like was there before) and using the instance locally there this would workaround things without resorting to accessing private methods and would be independent of any changes being done in Terra (unless I'm missing something). From an external perspective there would be no difference if in the future it did things as you show here directly on some internal cached instance though there maybe some performance benefit around reusing a cached instance on the same/similar problem.  So I can see there is room for improvement over what was there - indeed none of the current setters really working is less than ideal! Of course this code is not so tough to create your own factory. I imagine VQE will change as the expectation work gets further along - maybe at that point this can be revisited. Like the user passes in an ansatz I am wondering why we did not let them pass in a VQE instance too - that way they could set whatever they wanted on it and there would be no need to  expose a set of setter/getters that match VQE. Anyway if you want to go this way we need to remember to change these - maybe in additional to some todos you can raise a new issue here to fix it up and link it to the above Terra one as any changes will be dependent on that.